Cabal/src/Distribution/Simple/Errors.hs

```diff
@@ -353,3 +438,210 @@ exceptionMessage e = case e of
+          ++ ")"
+      reportFailedDependency (DependencyNoVersion dep) =
+        "cannot satisfy dependency " ++ prettyShow (simplifyDependency dep) ++ "\n"
+  NoPackageDatabaseSpecified ->
+    "No package databases have been specified. If you use "
+      ++ "--package-db=clear, you must follow it with --package-db= "
+      ++ "with 'global', 'user' or a specific file."
+  HowToFindInstalledPackages flv ->
+    "don't know how to find the installed packages for "
+      ++ prettyShow flv
+  PkgConfigNotFound pkg versionRequirement ->
+    "The pkg-config package '"
+      ++ pkg
+      ++ "'"
+      ++ versionRequirement
+      ++ " is required but it could not be found."
+  BadVersion pkg versionRequirement v ->
+    "The pkg-config package '"
+      ++ pkg
+      ++ "'"
+      ++ versionRequirement
+      ++ " is required but the version installed on the"
+      ++ " system is version "
+      ++ prettyShow v
+  UnknownCompilerException -> "Unknown compiler"
+  NoWorkingGcc ->
+    unlines
+      [ "No working gcc"
+      , "This package depends on foreign library but we cannot "
+          ++ "find a working C compiler. If you have it in a "
+          ++ "non-standard location you can use the --with-gcc "
+          ++ "flag to specify it."
+      ]
+  NoOSSupport os ->
+    "Operating system: "
+      ++ prettyShow os
+      ++ ", does not support relocatable builds"
+  NoCompilerSupport comp ->
+    "Compiler: "
+      ++ comp
+      ++ ", does not support relocatable builds"
+  InstallDirsNotPrefixRelative installDirs -> "Installation directories are not prefix_relative:\n" ++ show installDirs
+  ExplainErrors hdr libs ->
+    unlines $
+      [ if plural
+        then "Missing dependencies on foreign libraries:"
+        else "Missing dependency on a foreign library:"
+      | missing
+      ]
+        ++ case hdr of
+          Just (Left h) -> ["* Missing (or bad) header file: " ++ h]
+          _ -> []
+        ++ case libs of
+          [] -> []
+          [lib] -> ["* Missing (or bad) C library: " ++ lib]
+          _ ->
+            [ "* Missing (or bad) C libraries: "
+                ++ intercalate ", " libs
+            ]
+              ++ [if plural then messagePlural else messageSingular | missing]
+        ++ case hdr of
+          Just (Left _) -> [headerCppMessage]
+          Just (Right h) ->
+            [ (if missing then "* " else "")
+                ++ "Bad header file: "
+                ++ h
+            , headerCcMessage
+            ]
+          _ -> []
+    where
+      plural = length libs >= 2
+      -- Is there something missing? (as opposed to broken)
+      missing =
+        not (null libs)
+          || case hdr of Just (Left _) -> True; _ -> False
+      messageSingular =
+        "This problem can usually be solved by installing the system "
+          ++ "package that provides this library (you may need the "
+          ++ "\"-dev\" version). If the library is already installed "
```

Comment:
The suffix is "-devel" on RHEL and Fedora and derivatives. Dunno if it's worth notiing here.

Reply:
I am not sure either.
@gbaz Any comments if this to be noted as part of this PR?